src/collapse/dtrace.rs

```diff
@@ -0,0 +1,219 @@
+use super::Collapse;
+use smallvec::SmallVec;
+use std::collections::{HashMap, VecDeque};
+use std::io;
+use std::io::prelude::*;
+
+/// Settings that change how frames are named from the incoming stack traces.
+///
+/// All options default to off.
+#[derive(Clone, Debug, Default)]
+pub struct Options {
+    /// Include raw addresses (e.g., `0xbfff0836`) where symbols can't be found.
+    pub include_addrs: bool,
+
+    /// Annotate kernel functions with a `_[k]` suffix.
+    pub annotate_kernel: bool,
+}
+
+/// A stack collapser for the output of `perf script`.
+///
+/// To construct one, either use `perf::Folder::default()` or create an [`Options`] and use
+/// `perf::Folder::from(options)`.
+#[derive(Default)]
+pub struct Folder {
+    /// All lines until the next empty line are stack lines.
+    in_event: bool,
+
+    /// Skip all stack lines in this event.
+    skip_stack: bool,
+
+    /// Function entries on the stack in this entry thus far.
+    stack: VecDeque<String>,
+    count: usize,
+    /// Number of times each call stack has been seen.
+    occurrences: HashMap<String, usize>,
+
+    /// Current comm name.
+    ///
+    /// Called pname after original stackcollapse-perf source.
+    pname: Option<String>,
```

Comment:
I don't think this is set anywhere, so it should probably be removed?

Reply:
👍 removed